features/topology-map/plugins/org.opennms.features.topology.plugins.topo.linkd/src/main/java/org/opennms/features/topology/plugins/topo/linkd/internal/LinkdTopologyProvider.java

```diff
@@ -0,0 +1,929 @@
+    }
+
+    public LldpElementDao getLldpElementDao() {
+        return m_lldpElementDao;
+    }
+
+    public void setOspfLinkDao(OspfLinkDao ospfLinkDao) {
+        m_ospfLinkDao = ospfLinkDao;
+    }
+
+    public OspfLinkDao getOspfLinkDao(){
+        return m_ospfLinkDao;
+    }
+
+    public IsIsLinkDao getIsisLinkDao() {
+        return m_isisLinkDao;
+    }
+
+    public void setIsisLinkDao(IsIsLinkDao isisLinkDao) {
+        m_isisLinkDao = isisLinkDao;
+    }
+
+    public IsIsElementDao getIsisElementDao() {
+        return m_isisElementDao;
+    }
+
+    public void setIsisElementDao(IsIsElementDao isisElementDao) {
+        m_isisElementDao = isisElementDao;
+    }
+
+    public BridgeTopologyDao getBridgeTopologyDao() {
+        return m_bridgeTopologyDao;
+    }
+
+    public void setBridgeTopologyDao(BridgeTopologyDao bridgeTopologyDao) {
+        m_bridgeTopologyDao = bridgeTopologyDao;
+    }
+
+    public IpNetToMediaDao getIpNetToMediaDao() {
+        return m_ipNetToMediaDao;
+    }
+    
+    public void setIpNetToMediaDao(IpNetToMediaDao ipNetToMediaDao) {
+        m_ipNetToMediaDao = ipNetToMediaDao;
+    }
+
+    public CdpLinkDao getCdpLinkDao() {
+        return m_cdpLinkDao;
+    }
+
+    public void setCdpLinkDao(CdpLinkDao cdpLinkDao) {
+        m_cdpLinkDao = cdpLinkDao;
+    }
+
+    public CdpElementDao getCdpElementDao() {
+        return m_cdpElementDao;
+    }
+
+    public void setCdpElementDao(CdpElementDao cdpElementDao) {
+        m_cdpElementDao = cdpElementDao;
+    }
+        
+    @Override
+    public Defaults getDefaults() {
+        return new Defaults()
+                .withSemanticZoomLevel(Defaults.DEFAULT_SEMANTIC_ZOOM_LEVEL)
+                .withPreferredLayout("D3 Layout") // D3 Layout
+                .withCriteria(() -> {
+                    final OnmsNode node = m_topologyDao.getDefaultFocusPoint();
+
+                    if (node != null) {
+                        final Vertex defaultVertex = getVertex(TOPOLOGY_NAMESPACE_LINKD, node.getNodeId());
+                        if (defaultVertex != null) {
+                            return Lists.newArrayList(LinkdHopCriteria.createCriteria(node.getNodeId(), node.getLabel()));
+                        }
+                    }
+                    return Lists.newArrayList();
+                });
+    }
+    
+    @Transactional
+    private void doRefresh() {        
+        Timer.Context vcontext = m_loadIpInterfacesTimer.time();
+        Map<InetAddress, OnmsIpInterface>  ipToOnmsIpMap = new HashMap<InetAddress, OnmsIpInterface>();
+        Set<InetAddress> duplicated = new HashSet<InetAddress>();
+        try {
+            for (OnmsIpInterface ip: m_ipInterfaceDao.findAll()) {
+                if (!m_nodeToOnmsIpPrimaryMap.containsKey(ip.getNode().getId())) {
+                    m_nodeToOnmsIpPrimaryMap.put(ip.getNode().getId(), ip);
+                }
+                if (ip.getIsSnmpPrimary().equals(PrimaryType.PRIMARY)) {
```

Comment:
No because if it you have a multihomed host you need to map only the primary....There is no guaranteed that you have a primary ip instead while you want to assign an ip address to identify the vertex....